drafts/prometheus-alertmanager.adoc

```diff
@@ -0,0 +1,250 @@
+
+=== Set up Prometheus to communicate with Open Liberty
+JENNIFER: Add a sentence here saying what the reader will do here and to what end? where are these steps done (eg is the yaml file in Prometheus configuration?). eg "You need to configure Prometheus to communicate with the Open Liberty server in your app."
+
+. Edit the `prometheus.yml` file. Under the `scrape_configs` section, there is already a job configured for Prometheus itself. We need to add one for Open Liberty, which should look something like this:
++
+```
+- job_name: 'openliberty'
+
+    scheme: 'https'
+
+    tls_config:
+      insecure_skip_verify: true
+
+    static_configs:
+      - targets: ['localhost:9443']
+```
++
+. Run `./prometheus` (or `prometheus.exe` on Windows). When you see the message `Server is ready to receive web requests`, go to `http://localhost:9090` to view the Prometheus web UI.
+. To verify that your setup is working, click on `Targets` under `Status` in the navigation bar. You should see your `openliberty` target with the status `UP`.
+
+image::/img/blog/prometheusAM_ui_status.png[Prometheus Status on UI, align="left"]
+
+You're now ready to define which metrics you want Prometheus to collect.
+
+=== Creating Prometheus alerting rules
+
+Create some alerting rules in Prometheus to define which metrics will be received by Prometheus for passing on to Slack.
+
+. Make a file called `alert.yml`. (JENNIFER: Where?) Inside that file, create some alerting rules, for example:
++
+```
+groups:
+- name: libertyexample
+  rules:
+  - alert: heapUsageTooHigh
+    expr: base_memory_usedHeap_bytes / base_memory_maxHeap_bytes > 0.9
+    for: 1m
+    labels:
+      severity: warning
+    annotations:
+      summary: "Heap usage is too high"
+      description: "{{ $labels.instance }} heap usage is too high"
+```
++
+This rule will send an alert called `heapUsageTooHigh` when the PromQL query `base_memory_usedHeap_bytes / base_memory_maxHeap_bytes` is greater than `0.9` (90%) for one minute. (For testing purposes, feel free to change the threshold to something smaller, like `0.05`.)
++
+If you're new to PromQL, check out link:https://prometheus.io/docs/prometheus/latest/querying/basics/[Querying Prometheus].
++
+. In the `prometheus.yml` file, add the location of your `alert.yml` file under `rule_files`. If it's not in the same directory as Prometheus, make sure you supply the full path.
+. Restart Prometheus. Navigate to `http://localhost:9090` and click on `Alerts` on the navigation bar. You should be able to see your alert.
+
+image::/img/blog/prometheusAM_ui_alerts.png[Prometheus Alerts on UI, align="left"]
+
+== Configure Prometheus Alertmanager to pass alerts to Slack
+
+JENNIFER:  Add a brief introduction here to say what the reader will achieve/learn from this section (configure Prometheus Alertmanager to work with Slack and then  xxxx). 
```

Comment:
ADD: In this section, you'll configure the Prometheus Alertmanager to send customized messages to your Slack channel based on the alerting rules we set up earlier.